### ISSUE TITLE
PLR6104 false positive and false fix for augmented assignment.

### ISSUE THREAD

**Issue by LefterisJP** [astral-sh/ruff#10925]
### Problem

ruff 0.3.7

```python
a = 'le'
b = 'app'

a = b + a
print(a)
```

This prints `apple`.

If you run ruff it will hit with `PLR6104 Use `+=` to perform an augmented assignment directly` for line 4.

If you run the unsafe fix

```python3
a = 'le'
b = 'app'

a += b
print(a)
```

You will see that it tries to add b to a though what the previous code was doing is prefix a with b. This will now print `leapp`.

**Comment by AlexWaygood** [comment]
Thanks! I believe this is already fixed on `main`, so shouldn't be an issue in the next release :-)

Closing as a duplicate of #10911

**Comment by LefterisJP** [comment]
> Closing as a duplicate of #10911

Ah apologies Alex. I searched for PLR6104 but missed this one. Thanks for the quick response.
